lib/diplomat/value.ex

```diff
@@ -14,85 +19,102 @@ defmodule Diplomat.Value do
 
   def new(val), do: %__MODULE__{value: val}
 
-  def from_proto(%PbVal{boolean_value: val}) when is_boolean(val),
-    do: new(val)
-  def from_proto(%PbVal{integer_value: val}) when is_integer(val),
-    do: new(val)
-  def from_proto(%PbVal{double_value: val}) when is_float(val),
-    do: new(val)
-  def from_proto(%PbVal{string_value: val}) when is_binary(val),
-    do: new(val)
-  def from_proto(%PbVal{blob_value: val}) when is_bitstring(val),
-    do: new(val)
-  def from_proto(%PbVal{timestamp_microseconds_value: val}) when is_integer(val),
-    do: new(val) # need to convert this to a timestamp of some sort
-  def from_proto(%PbVal{key_value: val}) when not is_nil(val) do
+  def from_proto(%PbVal{value_type: {:boolean_value, val}}) when is_boolean(val) do
+    new(val)
+  end
+  def from_proto(%PbVal{value_type: {:integer_value, val}}) when is_integer(val) do
+    new(val)
+  end
+  def from_proto(%PbVal{value_type: {:double_value, val}}) when is_float(val) do
+    new(val)
+  end
+  def from_proto(%PbVal{value_type: {:string_value, val}}) when is_binary(val) do
+    new(val)
+  end
+  def from_proto(%PbVal{value_type: {:blob_value, val}}) when is_bitstring(val) do
+    new(val)
+  end
```

Comment:
I don't have a strong feeling about all function definitions having the same format. In fact, I think I prefer each definition be formatted according to what makes the most sense. For instance, something like the following is most appealing to me (it's a contrived example):

```elixir
def proto(nil), do: nil
def proto(val) when is_binary(val) do
  case String.valid?(val) do
    true -> PbVal.new(value_type: {:string_value, val})
    false-> PbVal.new(value_type: {:blob_value, val})
  end
end
```